docs/resources/live_recording.md

```diff
@@ -0,0 +1,128 @@
+---
+subcategory: "Live"
+---
+
+# huaweicloud_live_recording
+
+Manages a recording template within HuaweiCloud Live.
+
+## Example Usage
+
+### Create a recording template for an ingest domain name
+
+```hcl
+variable "ingest_domain_name" {}
+variable "bucket_region" {}
+variable "bucket_name" {}
+
+resource "huaweicloud_live_domain" "ingestDomain" {
+  name = var.ingest_domain_name
+  type = "push"
+}
+
+resource "huaweicloud_live_recording" "recording" {
+  domain_name = huaweicloud_live_domain.ingestDomain.name
+  app_name    = "live"
+  stream_name = "stream_name"
+  type        = "CONTINUOUS_RECORD"
+
+  obs {
+    region = var.bucket_region
+    bucket = var.bucket_name
+  }
+
+  hls {
+    recording_length = 15
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `domain_name` - (Required, String) Specifies the ingest domain name.
+
+* `app_name` - (Required, String) Specifies the application name. To match all names, use an asterisk (*).
+
+* `stream_name` - (Required, String) Specifies the stream name. To match all names, use an asterisk (*).
+
+* `type` - (Optional, String, ForceNew) Specifies the types of recording notifications. The options are as follows:
+  + **CONTINUOUS_RECORD**: continuous recording. Recording is triggered once streams are pushed to the recording system.
+  + **COMMAND_RECORD**: command-based recording. Tenants need to run commands to start and stop recording after streams
+   are pushed to the recording system.
+
+  Defaults to `CONTINUOUS_RECORD`. Changing this parameter will create a new resource.
+
+* `obs` - (Required, List) Specifies the obs for storing recordings.
+The [obs](#recording_obs) structure is documented below.
+
+* `hls` - (Required, List) Specifies the HLS configuration rule for storing recording as HLS.
+The [hls](#recording_HLS) structure isdocumented below.
+
+* `flv` - (Required, List) Specifies the FLV configuration rule for storing recording as FLV.
+The [flv](#recording_FLV_MP4) structure isdocumented below.
+
+* `mp4` - (Required, List) Specifies the MP4 configuration rule for storing recording as MP4.
+The [mp4](#recording_FLV_MP4) structure isdocumented below.
+
+-> At least one of `hls`, `flv`, `mp4` must be specified.
+
+<a name="recording_obs"></a>
+The `obs` block supports:
+
+* `region` - (Required, String) Specifies the region of OBS.
```

Comment:
just a question here, do we support to use obs in a region different with live recording? seems we don't open this to users on console side.

Reply:
this is  required parameter in API

Reply:
I know, but if it can only be same with the live_recording, we can help to set for users.

Reply:
The same region is the best, but there are no restrictions on users

Reply:
ok, fair enough